src/comments.c

```diff
@@ -175,60 +175,63 @@ comments_process_update_result (const struct updateResult * const result, gpoint
 	commentFeed->updateJob = NULL;
 
 	/* rerender item with new comments */
-	itemview_update_item (item); 
+	itemview_update_item (item);
 	itemview_update ();
-	
+
 	item_unload (item);
-		
+
 	debug_exit ("comments_process_update_result");
 }
 
 void
-comments_refresh (itemPtr item) 
-{ 
-	commentFeedPtr		commentFeed = NULL;
-	updateRequestPtr	request;
-	const gchar		*url;
-	
+comments_refresh (itemPtr item)
+{
+	commentFeedPtr	commentFeed = NULL;
+	UpdateRequest	*request;
+	const gchar	*url;
+
 	if (!network_monitor_is_online ())
 		return;
-		
+
 	if (metadata_list_get (item->metadata, "commentFeedGone")) {
 		debug0 (DEBUG_UPDATE, "Comment feed returned HTTP 410. Not updating anymore!");
 		return;
 	}
-	
+
 	url = metadata_list_get (item->metadata, "commentFeedUri");
 	if (url) {
 		debug2 (DEBUG_UPDATE, "Updating comments for item \"%s\" (comment URL: %s)", item->title, url);
 
-		// FIXME: restore update state from DB?		
-		
+		// FIXME: restore update state from DB?
+
 		if (item->commentFeedId) {
 			commentFeed = comment_feed_from_id (item->commentFeedId);
 		} else {
-			item->commentFeedId = node_new_id ();		
+			item->commentFeedId = node_new_id ();
 			db_item_update (item);
 		}
 
-		if (!commentFeed) {			
+		if (!commentFeed) {
 			commentFeed = g_new0 (struct commentFeed, 1);
 			commentFeed->id = g_strdup (item->commentFeedId);
 			commentFeed->itemId = item->id;
-			commentFeed->updateState = update_state_new ();
 
 			if (!commentFeeds)
 				commentFeeds = g_hash_table_new (g_str_hash, g_str_equal);
 			g_hash_table_insert (commentFeeds, commentFeed->id, commentFeed);
 		}
 
-		request = update_request_new ();
-		request->options = g_new0 (struct updateOptions, 1);	// FIXME: use copy of parent subscription options
-		request->source = g_strdup (url);
+		request = update_request_new (
+			url,
+			commentFeed->updateState,
```

Comment:
You removed `update_state_new` above, so here `commentFeed->updateState` could be uninitialized.

Reply:
This should be ok, as it is NULL in this case and update_request_new() creates a new one.

Together with an additional fix in the result processing callback, where I now explicitely copy the result update state, this should be fine.